core/arch/arm/plat-imx/psci.c

```diff
@@ -41,23 +41,14 @@
 #include <tee/entry_std.h>
 #include <tee/entry_fast.h>
 
-static vaddr_t src_base(void)
-{
-	static void *va;
-
-	if (cpu_mmu_enabled()) {
-		if (!va)
-			va = phys_to_virt(SRC_BASE, MEM_AREA_IO_SEC);
-		return (vaddr_t)va;
-	}
-	return SRC_BASE;
-}
-
 int psci_cpu_on(uint32_t core_idx, uint32_t entry,
 		uint32_t context_id __attribute__((unused)))
 {
 	uint32_t val;
-	vaddr_t va = src_base();
+	vaddr_t va = core_mmu_get_io_va(SRC_BASE);
```

Comment:
Complementary to discussions [above](https://github.com/OP-TEE/optee_os/pull/1570/files#diff-cf3af624ec25bbbb7a27757f81b1d921R71), here `register_phys_mem(SRC_BASE, ...);` is externally defined.

So if `SRC_BASE` is really expected secure, this generic **psci.c** should at least assert the secure state of the target ram. If the secure state does not matter, then fine.
And same remark, why not simply call a generic `core_mmu_get_va(SRC_BASE, MEM_AREA_IO_SEC)`?